// howlongtobeat-api/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Platform = exports.Genre = exports.Flow = exports.Perspective = exports.Modifier = exports.SortOrder = exports.SortBy = exports.LengthType = exports.get = exports.find = void 0;
var howlongtobeat_1 = require("./howlongtobeat");
Object.defineProperty(exports, "find", { enumerable: true, get: function () { return howlongtobeat_1.find; } });
Object.defineProperty(exports, "get", { enumerable: true, get: function () { return howlongtobeat_1.get; } });
var types_1 = require("./types");
Object.defineProperty(exports, "LengthType", { enumerable: true, get: function () { return types_1.LengthType; } });
Object.defineProperty(exports, "SortBy", { enumerable: true, get: function () { return types_1.SortBy; } });
Object.defineProperty(exports, "SortOrder", { enumerable: true, get: function () { return types_1.SortOrder; } });
Object.defineProperty(exports, "Modifier", { enumerable: true, get: function () { return types_1.Modifier; } });
Object.defineProperty(exports, "Perspective", { enumerable: true, get: function () { return types_1.Perspective; } });
Object.defineProperty(exports, "Flow", { enumerable: true, get: function () { return types_1.Flow; } });
Object.defineProperty(exports, "Genre", { enumerable: true, get: function () { return types_1.Genre; } });
Object.defineProperty(exports, "Platform", { enumerable: true, get: function () { return types_1.Platform; } });
exports.default = {
    find: howlongtobeat_1.find,
    get: howlongtobeat_1.get
};
