// howlongtobeat-api/dist/types.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Platform = exports.Genre = exports.Flow = exports.Perspective = exports.Modifier = exports.SortOrder = exports.SortBy = exports.LengthType = exports.PLATFORM_FIELDS = exports.SPEEDRUN_FIELDS = exports.RELATED_CONTENT_FIELDS = exports.MULTI_PLAYER_FIELDS = exports.SINGLE_PLAYER_FIELDS = void 0;
exports.SINGLE_PLAYER_FIELDS = [
    'type',
    'polled',
    'average',
    'median',
    'rushed',
    'leisure'
];
exports.MULTI_PLAYER_FIELDS = [
    'type',
    'polled',
    'average',
    'median',
    'least',
    'most'
];
exports.RELATED_CONTENT_FIELDS = [
    'name',
    'polled',
    'rated',
    'main',
    'extended',
    'completionist',
    'all'
];
exports.SPEEDRUN_FIELDS = [
    'type',
    'polled',
    'average',
    'median',
    'fastest',
    'slowest'
];
exports.PLATFORM_FIELDS = [
    'platform',
    'polled',
    'main',
    'extended',
    'completionist',
    'fastest',
    'longest'
];
/*
 * Some enums to check if input params are valid
 */
var LengthType;
(function (LengthType) {
    LengthType["Main"] = "main";
    LengthType["Extended"] = "mainp";
    LengthType["Completionist"] = "comp";
    LengthType["Average"] = "averagea";
})(LengthType = exports.LengthType || (exports.LengthType = {}));
var SortBy;
(function (SortBy) {
    SortBy["Name"] = "name";
    SortBy["Main"] = "main";
    SortBy["Extended"] = "mainp";
    SortBy["Completionist"] = "comp";
    SortBy["AverageTime"] = "averagea";
    SortBy["TopRated"] = "rating";
    SortBy["MostPopular"] = "popular";
    SortBy["MostBacklogs"] = "backlog";
    SortBy["MostSubmissiones"] = "usersp";
    SortBy["MostPlayed"] = "playing";
    SortBy["MostSpeedruns"] = "speedrun";
    SortBy["ReleaseDate"] = "release";
})(SortBy = exports.SortBy || (exports.SortBy = {}));
var SortOrder;
(function (SortOrder) {
    SortOrder[SortOrder["Ascending"] = 1] = "Ascending";
    SortOrder[SortOrder["Descending"] = 0] = "Descending";
})(SortOrder = exports.SortOrder || (exports.SortOrder = {}));
var Modifier;
(function (Modifier) {
    Modifier["None"] = "";
    Modifier["HideDLC"] = "hide_dlc";
    Modifier["OnlyDLC"] = "only_dlc";
})(Modifier = exports.Modifier || (exports.Modifier = {}));
var Perspective;
(function (Perspective) {
    Perspective["All"] = "";
    Perspective["FirstPerson"] = "First-Person";
    Perspective["Isometric"] = "Isometric";
    Perspective["Side"] = "Side";
    Perspective["Text"] = "Text";
    Perspective["ThirdPerson"] = "Third-Person";
    Perspective["TopDown"] = "Top-Down";
    Perspective["VirtualReality"] = "VirtualReality";
})(Perspective = exports.Perspective || (exports.Perspective = {}));
var Flow;
(function (Flow) {
    Flow["All"] = "";
    Flow["Incremental"] = "Incremental";
    Flow["MassivelyMutiplayer"] = "Massively Mutiplayer";
    Flow["Multidirectional"] = "Multidirectional";
    Flow["OnRails"] = "On-Rails";
    Flow["PointAndClick"] = "Point-and-Click";
    Flow["RealTime"] = "Real-Time";
    Flow["Scrolling"] = "Scrolling";
    Flow["TurnBased"] = "Turn-Based";
})(Flow = exports.Flow || (exports.Flow = {}));
var Genre;
(function (Genre) {
    Genre["All"] = "";
    Genre["Action"] = "Action";
    Genre["Adventure"] = "Adventure";
    Genre["Arcade"] = "Arcade";
    Genre["BattleArena"] = "Battle Arena";
    Genre["BeatEmUp"] = "Beat 'em Up";
    Genre["BoardGame"] = "Board Game";
    Genre["Breakout"] = "Breakout";
    Genre["CardGame"] = "Card Game";
    Genre["CityBuilding"] = "City-Building";
    Genre["Compilation"] = "Compilation";
    Genre["Educational"] = "Educational";
    Genre["Fighting"] = "Fighting";
    Genre["Fitness"] = "Fitness";
    Genre["Flight"] = "Flight";
    Genre["FMV"] = "Full Motion Video (FMV)";
    Genre["HackAndSlash"] = "Hack and Slash";
    Genre["HiddenObject"] = "Hidden Object";
    Genre["Horror"] = "Horror";
    Genre["InteractiveArt"] = "Interactive Art";
    Genre["Management"] = "Management";
    Genre["MusicRhythm"] = "Music/Rhythm";
    Genre["OpenWorld"] = "Open World";
    Genre["Party"] = "Party";
    Genre["Pinball"] = "Pinball";
    Genre["Platform"] = "Platform";
    Genre["Puzzle"] = "Puzzle";
    Genre["RacingDriving"] = "Racing/Driving";
    Genre["Roguelike"] = "Roguelike";
    Genre["RolePlaying"] = "Role-Playing";
    Genre["Sandbox"] = "Sandbox";
    Genre["Shooter"] = "Shooter";
    Genre["Simulation"] = "Simulation";
    Genre["Social"] = "Social";
    Genre["Sports"] = "Sports";
    Genre["Stealth"] = "Stealth";
    Genre["StrategyTactical"] = "Strategy/Tactical";
    Genre["Survival"] = "Survival";
    Genre["TowerDefense"] = "Tower Defense";
    Genre["Trivia"] = "Trivia";
    Genre["VehicularCombat"] = "Vehicular Combat";
    Genre["VisualNovel"] = "Visual Novel";
})(Genre = exports.Genre || (exports.Genre = {}));
var Platform;
(function (Platform) {
    Platform["All"] = "";
    Platform["ThreeDO"] = "3DO";
    Platform["AcornArchimedes"] = "Acorn Archimedes";
    Platform["Amiga"] = "Amiga";
    Platform["AmigaCD32"] = "Amiga CD32";
    Platform["AmstradCPC"] = "Amstrad CPC";
    Platform["Android"] = "Android";
    Platform["AppleII"] = "Apple II";
    Platform["Arcade"] = "Arcade";
    Platform["Atari8bit"] = "Atari 8-bit Family";
    Platform["Atari2600"] = "Atari 2600";
    Platform["Atari5200"] = "Atari 5200";
    Platform["Atari7800"] = "Atari 7800";
    Platform["AtariJaguar"] = "Atari Jaguar";
    Platform["AtariJaguarCD"] = "Atari Jaguar CD";
    Platform["AtariLynx"] = "Atari Lynx";
    Platform["AtariST"] = "Atari ST";
    Platform["BBCMicro"] = "BBC Micro";
    Platform["Browser"] = "Browser";
    Platform["ColecoVision"] = "ColecoVision";
    Platform["Commodore64"] = "Commodore 64";
    Platform["CommodorePET"] = "Commodore PET";
    Platform["CommodoreVIC20"] = "Commodore VIC-20";
    Platform["Dreamcast"] = "Dreamcast";
    Platform["Emulated"] = "Emulated";
    Platform["FMTowns"] = "FM Towns";
    Platform["GameAndWatch"] = "Game & Watch";
    Platform["GameBoy"] = "Game Boy";
    Platform["GameBoyAdvance"] = "Game Boy Advance";
    Platform["GameBoyColor"] = "Game Boy Color";
    Platform["GearVR"] = "Gear VR";
    Platform["Gizmondo"] = "Gizmondo";
    Platform["GoogleStadia"] = "Google Stadia";
    Platform["Intellivision"] = "Intellivision";
    Platform["InteractiveMovie"] = "Interactive Movie";
    Platform["iOS"] = "iOS";
    Platform["Linux"] = "Linux";
    Platform["Mac"] = "Mac";
    Platform["Mobile"] = "Mobile";
    Platform["MSX"] = "MSX";
    Platform["NGage"] = "N-Gage";
    Platform["NECPC88"] = "NEC PC-88";
    Platform["NECPC98"] = "NEC PC-98";
    Platform["NECPCFX"] = "NEC PC-FX";
    Platform["NeoGeo"] = "Neo Geo";
    Platform["NeoGeoCD"] = "Neo Geo CD";
    Platform["NeoGeoPocket"] = "Neo Geo Pocket";
    Platform["NES"] = "NES";
    Platform["Nintendo3DS"] = "Nintendo 3DS";
    Platform["Nintendo64"] = "Nintendo 64";
    Platform["NintendoDS"] = "Nintendo DS";
    Platform["NintendoGameCube"] = "Nintendo GameCube";
    Platform["NintendoSwitch"] = "Nintendo Switch";
    Platform["OculusGo"] = "Oculus Go";
    Platform["OculusQuest"] = "Oculus Quest";
    Platform["Odyssey"] = "Odyssey";
    Platform["Odyssey2"] = "Odyssey 2";
    Platform["OnLive"] = "OnLive";
    Platform["Ouya"] = "Ouya";
    Platform["PC"] = "PC";
    Platform["PCVR"] = "PC VR";
    Platform["PhilipsCDi"] = "Philips CD-i";
    Platform["PlayStation"] = "PlayStation";
    Platform["PlayStation2"] = "PlayStation 2";
    Platform["PlayStation3"] = "PlayStation 3";
    Platform["PlayStation4"] = "PlayStation 4";
    Platform["PlayStation5"] = "PlayStation 5";
    Platform["PlayStationMobile"] = "PlayStation Mobile";
    Platform["PlayStationNow"] = "PlayStation Now";
    Platform["PlayStationPortable"] = "PlayStation Portable";
    Platform["PlayStationVita"] = "PlayStation Vita";
    Platform["PlayStationVR"] = "PlayStation VR";
    Platform["PlugAndPlay"] = "Plug & Play";
    Platform["Sega32X"] = "Sega 32X";
    Platform["SegaCD"] = "Sega CD";
    Platform["SegaGameGear"] = "Sega Game Gear";
    Platform["SegaMasterSystem"] = "Sega Master System";
    Platform["SegaMegaDrive"] = "Sega Mega Drive/Genesis";
    Platform["SegaPico"] = "Sega Pico";
    Platform["SegaSaturn"] = "Sega Saturn";
    Platform["SG1000"] = "SG-1000";
    Platform["SharpX1"] = "Sharp X1";
    Platform["SharpX68000"] = "Sharp X68000";
    Platform["SuperNintendo"] = "Super Nintendo";
    Platform["TigerHandheld"] = "Tiger Handheld";
    Platform["TurboGrafx16"] = "TurboGrafx-16";
    Platform["TurboGrafxCD"] = "TurboGrafx-CD";
    Platform["VirtualBoy"] = "Virtual Boy";
    Platform["Wii"] = "Wii";
    Platform["WiiU"] = "Wii U";
    Platform["WindowsPhone"] = "Windows Phone";
    Platform["WonderSwan"] = "WonderSwan";
    Platform["Xbox"] = "Xbox";
    Platform["Xbox360"] = "Xbox 360";
    Platform["XboxOne"] = "Xbox One";
    Platform["XboxSeries"] = "Xbox Series X/S";
    Platform["Zeebo"] = "Zeebo";
    Platform["ZXSpectrum"] = "ZX Spectrum";
})(Platform = exports.Platform || (exports.Platform = {}));
